static/app/views/eventsV2/table/columnEditCollection.tsx

```diff
@@ -120,11 +138,25 @@ class ColumnEditCollection extends React.Component<Props, State> {
 
   handleUpdateColumn = (index: number, column: Column) => {
     const newColumns = [...this.props.columns];
+    if (column.kind === 'equation') {
+      this.setState(prevState => {
+        const error = prevState.error;
+        error.set(index, parseArithmetic(column.field).error);
+        return {
+          ...prevState,
+          error,
+        };
+      });
```

Comment:
This feels like incorrect react as you're mutating the same map when setting the state. I believe the only reason why this is working is because the `onChange` below is causing the props to change which is causing the re-render.

Reply:
Hmm even with the onChange commented out the error state still updates

Reply:
I think you might need to clone the map. i.e. `newErrors = new Map(error)`

Reply:
🙏 sweet updated